examples/hetero/hetero_unsupervised_dblp.py

```diff
@@ -0,0 +1,187 @@
+import numpy as np
+from torch.nn import Parameter
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LogisticRegression
+import os.path as osp
+import math
+import torch
+import torch.nn.functional as F
+from torch_geometric.datasets import DBLP
+from torch_geometric.nn import SAGEConv, HeteroConv, GATConv
+
+EPS = 1e-15
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '../../data/DBLP')
+dataset = DBLP(path)
+data = dataset[0]
+print(data)
+
+# We initialize conference node features with a single feature.
+data['conference'].x = torch.ones(data['conference'].num_nodes, 1)
+
+
+def uniform(size, tensor):
+    if tensor is not None:
+        bound = 1.0 / math.sqrt(size)
+        tensor.data.uniform_(-bound, bound)
+
+
+class GATEncoder(torch.nn.Module):
+    def __init__(self, in_channels, out_channels, heads):
+        super(GATEncoder, self).__init__()
+        self.conv = GATConv(in_channels, out_channels, heads)
+        self.activation = torch.nn.PReLU()
+
+    def forward(self, x: torch.Tensor, edge_index: torch.Tensor):
+        return self.activation(self.conv(x, edge_index))
+
+
+class HeteroUnsupervised(torch.nn.Module):
+    def __init__(self, metadata, hidden_channels, out_channels, num_layers):
+        super().__init__()
+
+        # for discriminator
+        self.embed_size = out_channels
+        self.weight = Parameter(torch.Tensor(out_channels, out_channels))
+
+        # for encoder
+        self.encoder = GATEncoder(hidden_channels, out_channels, heads=1)
+
+        self.convs = torch.nn.ModuleList()
+        for _ in range(num_layers):
+            conv = HeteroConv({
+                edge_type: SAGEConv((-1, -1), hidden_channels)
+                for edge_type in metadata[1]
+            })
+            self.convs.append(conv)
+
+    def forward(self, xx_dict, edge_index_dict):
```

Comment:
```suggestion
    def forward(self, x_dict, edge_index_dict):
```

Reply:
I particularly keep the code `xx_dict` because if I wrote `x_dict`, everything inside `x_dict` would change including `feature.shape`, and that is not what it ought to be.

Reply:
Isn't x_dict supposed to change after each convolution layer? Each convolution layer works on the output of the last convolution layer? In your current setup each layer works on the same unchanging input xx_dict